src/resultsdisassemblypage.cpp

```diff
@@ -0,0 +1,153 @@
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "resultsdisassemblypage.h"
+#include "ui_resultsdisassemblypage.h"
+
+#include <QMenu>
+#include <QFileDialog>
+#include <QFile>
+#include <QTextStream>
+#include <QMessageBox>
+#include <QString>
+#include <QListWidgetItem>
+#include <QProcess>
+#include <QDebug>
+#include <QDir>
+#include <QDirIterator>
+
+#include <KRecursiveFilterProxyModel>
+
+#include "parsers/perf/perfparser.h"
+#include "resultsutil.h"
+
+#include "models/costdelegate.h"
+#include "models/hashmodel.h"
+#include "models/topproxy.h"
+#include "models/treemodel.h"
+
+#include <QStandardItemModel>
+
+ResultsDisassemblyPage::ResultsDisassemblyPage(FilterAndZoomStack *filterStack, PerfParser *parser, QWidget *parent)
+        : QWidget(parent)
+        , ui(new Ui::ResultsDisassemblyPage)
+{
+    ui->setupUi(this);
+    ui->splitter->setStretchFactor(0, 1);
+    ui->splitter->setStretchFactor(1, 1);
+    ui->splitter->setStretchFactor(2, 8);
+}
+
+ResultsDisassemblyPage::~ResultsDisassemblyPage() = default;
+
+void ResultsDisassemblyPage::clear()
+{
+    if (ui->asmView->model() == nullptr)
+        return;
+
+    int rowCount = ui->asmView->model()->rowCount();
+    if (rowCount > 0) {
+        ui->asmView->model()->removeRows(0, rowCount, QModelIndex());
+    }
+}
+
+void ResultsDisassemblyPage::showDisassembly()
+{
+    // Show empty tab when selected symbol is not valid
+    if (m_curSymbol.symbol.isEmpty()) {
+        clear();
+        return;
+    }
+
+    // Call objdump with arguments: addresses range and binary file
+    QString processName = m_objdump;
+    QStringList arguments;
+    arguments << QLatin1String("-d") << QLatin1String("--start-address") << QLatin1String("0x") + QString::number(m_curSymbol.relAddr, 16) <<
+            QLatin1String("--stop-address") << QLatin1String("0x") + QString::number(m_curSymbol.relAddr + m_curSymbol.size, 16) <<
+            m_curSymbol.path;
+
+    showDisassembly(processName, arguments);
+}
+
+void ResultsDisassemblyPage::showDisassembly(QString processName, QStringList arguments)
+{
+    QTemporaryFile m_tmpFile;
+    QProcess asmProcess;
+
+    if (m_tmpFile.open()) {
+        asmProcess.start(processName, arguments);
+
+        if (!asmProcess.waitForStarted() || !asmProcess.waitForFinished()) {
+            return;
+        }
+        QTextStream stream(&m_tmpFile);
+        stream << asmProcess.readAllStandardOutput();
+        m_tmpFile.close();
+    }
+
+    if (m_tmpFile.open()) {
+        int row = 0;
+        QStandardItemModel *model = new QStandardItemModel();
```

Comment:
you are leaking the model, pass a parent:

auto *model = new QStandardItemModel(this);

better yet make it a member and clear it instead of creating a new model all the time for every time this function is called

Reply:
I have fixed this in later commit 16ac4aa6089c4c4b165f488ceff9b063a5880ba3